certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -2378,6 +2403,226 @@ def enable_autohsts(self, _unused_lineage, domains):
         # Save the current state to pluginstorage
         self._autohsts_save_state()
 
```

Comment:
You don't have to do this as it's not something we normally do and it's a fair bit of work, but while reviewing this PR I had a lot of questions. Some of them were answered eventually after reading/testing the code while others I added as a comment on this PR.

One thing that might be useful for other developers in the future and even ourselves now while trying to remember the details of this ancient PR would be to document how this all works technically. How the are OCSP directories used? What is the layout of the `_ocsp_prefetch` object in `PluginStorage`? How does Apache use the OCSP database? When does it pick up changes? Are there any race conditions to worry about?

If you wanted to add this documentation, I would personally just add it as a big inline comment above all the OCSP stuff. If you move this code to a standalone class as I suggested in another comment, it could be in the class docstring. You could also document some of this stuff closer to the relevant code if you think that makes more sense.